### ISSUE TITLE
var-naming standards include capital letters; capital letters trigger linting errors.

### ISSUE THREAD

**Issue by r-pufky** [ansible-community/ansible-lint#1812]
### Summary
Originally submitted to `ansible` for clarification, see here: https://github.com/ansible/ansible/issues/76748. Comments from the ansible folks are that ansible supports `[a-zA-Z0-9_]` for valid var naming conventions.

[Current documentation](https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html#creating-valid-variable-names) only specifies the required used of letters, numbers, and underscores. Without explicitly stating letter case, I assume it is validated against:  `[a-zA-Z0-9_]` (alphanumerics + _).

Depending on what the ansible var naming convention intent is, this should be clarified to include **lowercase** letters or fixed to not trigger linting errors on vars with Caps `[a-zA-Z0-9_]`.

Note: this was discovered as some role variables were used that mimicked config vars for an install application (e.g. {ROLE}_{APP_CONFIG_VAR}).

### Issue Type

Bug Report

### Component Name

ansible-lint

### Ansible Version

```console
$ ansible --version
ansible [core 2.12.1]
  config file = /home/{REDACTED}/git/srv/cfg/ansible/ansible.cfg
  configured module search path = ['/home/{REDACTED}/git/srv/cfg/ansible/.ansible/working/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/{REDACTED}/git/srv/cfg/ansible/.ansible/lib/python3.10/site-packages/ansible
  ansible collection location = /home/{REDACTED}/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/{REDACTED}/git/srv/cfg/ansible/.ansible/bin/ansible
  python version = 3.10.1 (main, Dec 18 2021, 23:53:45) [GCC 11.1.0]
  jinja version = 3.0.3
  libyaml = True
```

### Configuration

```console
$ ansible-config dump --only-changed
DEFAULT_HOST_LIST(/home/{REDACTED}/git/srv/cfg/ansible/ansible.cfg) = ['/home/{REDACTED}/git/srv/core/hosts']
DEFAULT_LOCAL_TMP(/home/{REDACTED}/git/srv/cfg/ansible/ansible.cfg) = /home/{REDACTED}/git/srv/cfg/ansible/.ansible/working/tmp/ansible-local-5758zava2h1j
DEFAULT_MODULE_PATH(/home/{REDACTED}/git/srv/cfg/ansible/ansible.cfg) = ['/home/{REDACTED}/git/srv/cfg/ansible/.ansible/working/plugins/modules', '/usr/share/ansible/plugins/modules']
DEFAULT_MODULE_UTILS_PATH(/home/{REDACTED}/git/srv/cfg/ansible/ansible.cfg) = ['/home/{REDACTED}/git/srv/cfg/ansible/.ansible/working/plugins/module_utils', '/usr/share/ansible/plugins/module_utils']
DEFAULT_PRIVATE_KEY_FILE(/home/{REDACTED}/git/srv/cfg/ansible/ansible.cfg) = /home/{REDACTED}/git/srv/cfg/ssh/core
DEFAULT_REMOTE_USER(/home/{REDACTED}/git/srv/cfg/ansible/ansible.cfg) = ansible
DEFAULT_STDOUT_CALLBACK(/home/{REDACTED}/git/srv/cfg/ansible/ansible.cfg) = debug
DEFAULT_VAULT_PASSWORD_FILE(/home/{REDACTED}/git/srv/cfg/ansible/ansible.cfg) = /home/{REDACTED}/git/srv/cfg/ansible/vault-gpg
DISPLAY_SKIPPED_HOSTS(/home/{REDACTED}/git/srv/cfg/ansible/ansible.cfg) = False
HOST_KEY_CHECKING(/home/{REDACTED}/git/srv/cfg/ansible/ansible.cfg) = True
INTERPRETER_PYTHON(/home/{REDACTED}/git/srv/cfg/ansible/ansible.cfg) = auto_silent
```

### OS / Environment

``` bash
$ cat /etc/lsb-release
DISTRIB_ID=ManjaroLinux
DISTRIB_RELEASE=21.2.1
DISTRIB_CODENAME=Qonos
DISTRIB_DESCRIPTION="Manjaro Linux"
```

### Steps to Reproduce

create a yaml file with a capped variable:

`test_var_naming_lint_error.yml`
```yaml
---
test_Captial: 'test var-naming lint error'
```

run `ansible-lint`

``` bash
$ ansible-lint test_var_naming_lint_error.yml
```

### Expected Results

If `ansible-lint` is supposed to validate `[a-zA-Z0-9_]`, then **no linting error** is expected.

Alternatively maybe there should be a "loose" option which disables Capped errors while still validating the rest of var naming tests.

~If `ansible-lint` is supposed to validate `[a-z0-9_]`, then [documentation](https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html#creating-valid-variable-names) should be updated.~ Ansible folks have confirmed that `[a-zA-Z0-9_]` is valid.

### Actual Results

```console
Currently running `ansible-lint` produces the following linting error:
 bash
$ ansible-lint test_var_naming_lint_error.yml
WARNING  Listing 1 violation(s) that are fatal
var-naming: File defines variable 'test_Captial' that violates variable naming standards
test_var_naming_lint_error.yml.yml:1
```

`test_var_naming_lint_error.yml`
``` yaml
---
test_Captial: 'test var-naming lint error'
```
```


### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by r-pufky** [comment]
> A `# noqa var-naming` at the end of the respective line helped to resolve that case.

Yup -- I currently worked around it by `noqa`'ing it. However what I am getting at is:

I'd like the ability to still apply the rest of the var-naming rule to the line for validation, specifically being able to disable just that  error; disabling it for the line also disables the rest of the checks that rule does (so maybe a custom submodule exclusion for the line or config file?) Only var-naming in this example, but could apply to all modules with multi checks for validation of that rule.

``` yaml
test_Capital: 'some var' # noqa var-naming.capitals
```

This very clearly describes in a programmatic fashion what is being disabled without the need for additional comments. It also makes it inherently extensible:

``` yaml
test-no-unders: 'some var' # noqa var-naming.underscores

- ansible.builtin.set_fact:
    'git_{{ package.name|replace("-","_") }}': '{{ git_changed }}' # noqa var-naming.standards
```

... and can be defined in the lint configuration file:
``` yaml
skip_list:
  - var-naming.captials
```

**Comment by ssbarnea** [comment]
We just had a talk within @ansible-community/devtools team and we decided that this would be a for the moment a wontfix, especially because this would likely prevent a collection from being certified.

If you really want to ignore that rule, you can add it to skip-list.
